tests/interfaces/test_tf.py

```diff
@@ -768,3 +768,379 @@ def test_model_nested_tensor(self):
             [u.numpy().shape == model[i].numpy().shape for i, u in enumerate(unflattened[0])]
         )
         assert unflattened[1].numpy().size == 0
+
+
+class TestParameterHandlingIntegration:
+    """Test that the parameter handling for differentiable/non-differentiable
+    parameters works correctly."""
+
+    def test_differentiable_parameter_first(self, mocker):
+        """Test that a differentiable parameter used as the first
+        argument is correctly evaluated by QNode.jacobian, and that
+        all other non-differentiable parameters are ignored"""
+        spy = mocker.spy(qml.qnodes.JacobianQNode, "jacobian")
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(weights, data1, data2):
+            # non-differentiable quantum function
+            qml.templates.AmplitudeEmbedding(data1, wires=[0, 1])
+            # differentiable quantum function
+            qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
+            # non-differentiable quantum function
+            qml.templates.AngleEmbedding(data2, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        # differentiating the circuit wrt the weights
+        # input weights
+        weights = qml.init.strong_ent_layers_normal(n_wires=2, n_layers=3)
+        weights = tf.Variable(weights)
+
+        # input data
+        data1 = tf.constant([0, 1, 1, 0], dtype=tf.float64) / np.sqrt(2)
+        data2 = tf.Variable([1, 1], dtype=tf.float64, trainable=False)
+
+        with tf.GradientTape() as tape:
+            loss = circuit(weights, data1, data2)
+
+        grad = tape.gradient(loss, weights)
+        res = grad.numpy()
```

Comment:
How come we need to convert to numpy? We can't check using tensors?

Reply:
Good catch; I've removed the superfluous numpy conversions.